File: ruby/muse/util/SequenceBuilder.rb
require "muse/util/RandomFactory"
require "muse/util/Sequences"

module SequenceBuilder

    def self.packIncreasing(arr)
        RandomFactory.seed()
        arr[0] = RandomFactory.genIntN(1, 3)
        1.upto(arr.length - 1) do |i|
            arr[i] = arr[i - 1] + RandomFactory.genIntN(1, 3)
        end
    end

    def self.packRandom(arr)
        RandomFactory.seed()
        arr.each_index do |i|
            arr[i] = RandomFactory.genInt()
        end
    end

    def self.packDecreasing(arr)
        packIncreasing(arr)
        Sequences.reverse(arr)
    end
end

File: nodejs/muse/util/SequenceBuilder.js
'use strict'

const RandomFactory = require('muse/util/RandomFactory')
const Sequences = require('muse/util/Sequences')

exports.packIncreasing = (arr) => {
  arr[0] = RandomFactory.genIntN(1, 3)
  for (let i = 1, length = arr.length; i < length; i++) {
    arr[i] = arr[i - 1] + RandomFactory.genIntN(1, 3)
  }
}

exports.packRandom = (arr) => {
  for (let i = 0, length = arr.length; i < length; i++) {
    arr[i] = RandomFactory.genInt()
  }
}

exports.packDecreasing = (arr) => {
  this.packIncreasing(arr)
  Sequences.reverse(arr)
}
